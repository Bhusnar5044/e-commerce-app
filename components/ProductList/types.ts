import { Product } from '@utils/types';

export interface IProps {
    title: string;
    items: Product[];
}
