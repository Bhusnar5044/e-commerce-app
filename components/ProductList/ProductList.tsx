import NoResults from '@/components/ui/NoResults';
import ProductCard from '@/components/ui/ProductCard';
import { memo } from 'react';
import SearchBox from '../SearchBox';
import SortFilter from '../SortFilter';
import { IProps } from './types';

export const ProductList: React.FC<IProps> = memo(({ title, items }) => {
    return (
        <div className="space-y-4">
            <div className="flex flex-wrap justify-between items-center sticky top-[4rem] z-30 bg-white">
                <h3 className="font-bold text-3xl">{title}</h3>
                <div className="flex gap-2 items-center">
                    <SearchBox />
                    <SortFilter />
                </div>
            </div>
            {items.length === 0 && <NoResults />}
            <div className="grid grid-cols-1 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4 gap-4">
                {items.map((item) => (
                    <ProductCard key={item.title} data={item} />
                ))}
            </div>
        </div>
    );
});

ProductList.displayName = 'ProductList';
