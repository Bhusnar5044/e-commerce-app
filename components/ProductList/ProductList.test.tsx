import { render } from '@test-utils';
import { ProductList } from './ProductList';

describe('ProductList', () => {
    it('should render properly', () => {
        const { container } = render(<ProductList title={'Dummy'} items={[]} />);
        expect(container).toMatchSnapshot();
    });
});
