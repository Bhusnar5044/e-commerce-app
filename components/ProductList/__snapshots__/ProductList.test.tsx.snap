// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductList should render properly 1`] = `
<div>
  <div
    class="space-y-4"
  >
    <div
      class="flex justify-between items-center sticky top-[4rem] z-30 bg-white"
    >
      <h3
        class="font-bold text-3xl"
      >
        Dummy
      </h3>
      <div
        class="flex gap-2 items-center"
      >
        <label
          class="mb-2 text-sm font-medium text-gray-900 sr-only dark:text-white"
          for="default-search"
        >
          Search
        </label>
        <div
          class="relative"
        >
          <input
            class="block w-full p-2 pl-4 text-sm text-gray-900 border border-gray-300 rounded-lg focus:ring-blue-500 focus:border-blue-500  dark:border-gray-600 dark:placeholder-gray-400 dark:focus:ring-blue-500 dark:focus:border-blue-500"
            id="default-search"
            placeholder="Search Mockups, Logos..."
            required=""
            type="search"
            value=""
          />
          <div
            class="absolute inset-y-1 right-0 flex items-center pr-3 pointer-events-cursor"
          >
            <svg
              aria-hidden="true"
              class="w-4 h-4 text-gray-500 dark:text-gray-400"
              fill="none"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m19 19-4-4m0-7A7 7 0 1 1 1 8a7 7 0 0 1 14 0Z"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </div>
        </div>
        <div
          class="flex items-center justify-center py-2"
        >
          <div
            class="relative inline-block text-left"
          >
            <span
              class="rounded-md shadow-sm"
            >
              <button
                aria-expanded="false"
                aria-haspopup="menu"
                class="inline-flex justify-center w-full px-4 py-2 text-sm font-medium leading-5 text-gray-700 transition duration-150 ease-in-out bg-white border border-gray-300 rounded-md hover:text-gray-500 focus:outline-none focus:border-blue-300 focus:shadow-outline-blue "
                data-headlessui-state=""
                id="headlessui-menu-button-:r0:"
                type="button"
              >
                <span>
                  Sort
                </span>
                <svg
                  class="w-5 h-5 ml-2 -mr-1"
                  fill="currentColor"
                  viewBox="0 0 20 20"
                >
                  <path
                    clip-rule="evenodd"
                    d="M5.293 7.293a1 1 0 011.414 0L10 10.586l3.293-3.293a1 1 0 111.414 1.414l-4 4a1 1 0 01-1.414 0l-4-4a1 1 0 010-1.414z"
                    fill-rule="evenodd"
                  />
                </svg>
              </button>
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      class="flex items-center justify-center h-full w-full text-neutral-500"
    >
      No results found.
    </div>
    <div
      class="grid grid-cols-1 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4 gap-4"
    />
  </div>
</div>
`;
